core/container.go

```diff
@@ -1644,6 +1677,34 @@ func (container *Container) Service(ctx context.Context) (*Service, error) {
 	return container.Query.NewContainerService(ctx, container), nil
 }
 
+// TODO(rajatjindal): bikeshedding for name
+func (container *Container) AsService(ctx context.Context, args ContainerAsServiceArgs) (*Service, error) {
```

Comment:
I think we tend to rename the current one to `LegacyXXX` (`LegacyService` & `Service`) if it's kept around just for compat reasons. Makes clear why there's two versions. Sometimes you can use the same function for old and new.